### ISSUE TITLE
Deprecation in wsdl.cs

### ISSUE THREAD

**Issue by UweKeim** [oleg-shilo/cs-script#45]
Using the latest version 3.23.0.0, I get a warning/error in the "wsdl.cs" that ships within the extension pack:

Error: Specified file could not be compiled.

> System.ApplicationException: Script \\myserver\myshare\cs-script\lib\wsdl.cs cannot be executed. ---> csscript.CompilerException: \\myserver\myshare\cs-script\lib\wsdl.cs(95,70): error CS0619: 'csscript.CSSEnvironment.PrimaryScriptFile' is obsolete: 'This member may not work correctly in the hosted and cached scenarios. Use alternative techniques demonstrated in the ReflectScript.cs sample.Including environment variable 'EntryScript' and AssemblyDescriptionAttribute (of the script assembly) containing the full path of the script file.'

The "wsdl.cs" file is back from 2009.

In addition, I am unable to locate the "wsdl.cs" file [in your repository](https://github.com/oleg-shilo/cs-script/find/master).

**My question:**

Can you please help me to get "wsdl.cs" working with the latest version of CS Script?

**Comment by UweKeim** [comment]
My current workaround is to use

```c#
new System.Diagnostics.StackTrace(new System.Diagnostics.StackFrame(true)).GetFrame(0).GetFileName()
```

as a replacement for 

```c#
CSSEnvironment.PrimaryScriptFile
```

(Idea from [here](https://groups.google.com/forum/#!searchin/cs-script/PrimaryScriptFile%7Csort:relevance/cs-script/_a7vcJ-S9vI/iOxneZe0xgAJ))

This seems to work.

Still, an updated extension pack would be awesome.

**Comment by oleg-shilo** [comment]
I am a day away from making a new release. So txs. Your feedback is just in time. The new release will have the new wsdl.cs.

The StackFrame may not always work you should use the direct approach (as in ReflectScript.cs):
```C#
static string ResolveOutputLocation(string file)
{
    string primaryScriptFile = Environment.GetEnvironmentVariable("location:" + Assembly.GetExecutingAssembly().GetHashCode());

    if (primaryScriptFile != null && CSScript.GlobalSettings.HideAutoGeneratedFiles == Settings.HideOptions.HideAll)
        return Path.Combine(CSSEnvironment.GetCacheDirectory(primaryScriptFile), Path.GetFileName(file));
    else if (primaryScriptFile != null)
        return Path.Combine(Path.GetDirectoryName(primaryScriptFile), file);
    else
        return Path.GetFullPath(file);
}
```

**Comment by UweKeim** [comment]
Thanks a lot, Oleg!

Maybe you could put the extension back here on GitHub, too?

**Comment by oleg-shilo** [comment]
Yes I will. This is something that I just overlooked.

BTW, my early suggestion did not pass the test :( 
Will post the revised answer after the investigation.

**Comment by oleg-shilo** [comment]
Hi Uwe,
Done. Please find the fix in the release v3.24.0

My initial suggestion was incorrect as it would return the location of the secondary (pre-script) but not the primary (entry) script. The correct env variable is `EntryScript`:
```C#
static string ResolveOutputLocation(string file)
{
    string primaryScriptFile = Environment.GetEnvironmentVariable("EntryScript");
    ...
}
```
Though the prev release had a problem - this variable was set after pre-script execution, what was too late for your purpose. This defect has also been fixed and the updated wsdl.cs can be found in the ExtensionPack.

And as as we agreed the all samples/lib distributable sources are now under source control: https://github.com/oleg-shilo/cs-script/tree/master/Source/deployment

Cheers,
Oleg

**Comment by UweKeim** [comment]
Awesome as always. Thanks a lot, Oleg.
